### PR TITLE
Simplify walkthrough steps

### PR DESCRIPTION
**Pull request by Eskibear**
To address #679

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,119 +55,69 @@
...
+    "onCommand:java.formatterSettings",
+    "onWalkthrough:javaWelcome"
   ],
   "contributes": {
     "walkthroughs": [
       {
         "id": "javaWelcome",
         "title": "Dive into Java Development",
         "description": "Your first steps to set up powerful Java tools in a lightweight, performant editor!",
-        "when": "false",
```
Comment (Eskibear) [inline package.json:66]: always enable it as we already finished the EXP. /cc @jdneo

File: package.json [hunk 2]
```diff
@@ -55,119 +55,69 @@
...
-            "when": "isMac && workspaceFolderCount == 0"
-          },
           {
             "id": "java.showProjectExplorer",
-            "title": "Explore your project",
-            "description": "View classpaths and dependencies, and create new projects, packages and classes.\n[Show Java Projects](command:javaProjectExplorer.focus)",
+            "title": "Explore your projects",
+            "description": "%walkthrough.java.projects%",
             "media": {
               "image": "walkthrough/project-manager.png",
```
Comment (Eskibear) [inline package.json:81]: TODO: needs a themed SVG

File: package.json [hunk 3]
```diff
@@ -55,119 +55,69 @@
...
-            "id": "java.showDebugView",
-            "title": "Launch and debug your app",
-            "description": "The debugger will automatically find the entry point of your project, launch it and start debugging.\n[Show Run and Debug View](command:workbench.view.debug)\n[Learn More...](https://code.visualstudio.com/docs/java/java-debugging)",
+            "id": "java.debugAndTest",
+            "title": "Launch, debug and test",
+            "description": "%walkthrough.java.debugAndTest%",
             "media": {
-              "image": "walkthrough/debugger.png",
-              "altText": "Running and Debugging"
+              "image": "walkthrough/debugAndTest.png",
```
Comment (Eskibear) [inline package.json:101]: placeholder, TODO: needs a themed SVG

File: package.json [hunk 4]
```diff
@@ -55,119 +55,69 @@
...
-            "id": "java.tutorial",
-            "title": "Quick start with Java",
-            "description": "Just starting with Java? Get hands on with these tutorials to learn the fundamentals of Java editing & debugging.\n[Open Tutorials](command:toSide:java.gettingStarted)",
+            "id": "java.extensions",
+            "title": "Additional tools and frameworks",
+            "description": "%walkthrough.java.extensions%",
             "media": {
-              "image": "walkthrough/tutorials.png",
-              "altText": "Tutorials"
+              "svg": "walkthrough/extensions.svg",
```
Comment (Eskibear) [inline package.json:110]: this SVG is pending modification, grouping extensions in categories., e.g. keymaps

File: package.json [hunk 5]
```diff
@@ -55,119 +55,69 @@
...
-            "id": "java.showHelpCenter",
-            "title": "Java Help Center",
-            "description": "Discover more shortcuts and features on this page.\n[Open Java Help Center](command:toSide:java.welcome.fromWalkthrough)",
+            "id": "java.more",
+            "title": "Explore more Java resources",
+            "description": "%walkthrough.java.more%",
             "media": {
-              "image": "walkthrough/help-center.png",
-              "altText": "Java Help Center"
+              "image": "walkthrough/more.png",
```
Comment (Eskibear) [inline package.json:119]: TODO: needs a themed SVG

File: walkthrough/runtime.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+### Install a JDK
+
+- Explain why to install JDK
```
Comment (Eskibear) [inline walkthrough/runtime.md:3]: need to enrich the content of this markdown file. cc @nickzhums

File: package.json [hunk 7]
```diff
@@ -55,119 +55,69 @@
...
-              "altText": "Testing"
-            },
-            "when": "!java:testRunnerActivated"
-          },
-          {
-            "id": "java.tutorial",
-            "title": "Quick start with Java",
-            "description": "Just starting with Java? Get hands on with these tutorials to learn the fundamentals of Java editing & debugging.\n[Open Tutorials](command:toSide:java.gettingStarted)",
+            "id": "java.extensions",
+            "title": "Additional tools and frameworks",
```
Comment (nickzhums) [inline package.json:107]: Should we say "Extensions for additional tools and frameworks" to make it more obvious?

File: walkthrough/runtime.md (hunk 6 above)
Comment (nickzhums) [reply walkthrough/runtime.md:3]: what's the main for this content again?

File: package.json (hunk 7 above)
Comment (Eskibear) [reply package.json:107]: true.

File: walkthrough/runtime.md (hunk 6 above)
Comment (Eskibear) [reply walkthrough/runtime.md:3]: For this step, the main purpose is o 
- provide a way to install new jdk
- let users know that at least one Java runtime is required.

So I'm not sure what content to put it here. Currently I prefer to provide a markdown with some descriptions just like python's?

File: package.nls.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
+    "walkthrough.java.runtime": "Make sure you have at least on Java runtime installed.\n[Install New JDK](command:toSide:java.installJdk.fromWalkthrough)",
```
Comment (nickzhums) [inline package.nls.json:2]: typo:one

The extension pack for Java requires at least one Java runtime to be installed. Install new JDK from here

File: package.nls.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+    "walkthrough.java.projects": "In Java Projects explorer, you can browse your projects, view classpaths and dependencies. \n[View Java Projects](command:javaProjectExplorer.focus)",
```
Comment (nickzhums) [inline package.nls.json:4]:
```suggestion
    "walkthrough.java.projects": "Java Project Explorer helps you to manage your projects and dependencies. It also helps you to create new Java project, classes and packages. \n[View Java Projects](command:javaProjectExplorer.focus)",
```

File: package.nls.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+    "walkthrough.java.projects.inactive": "In Java Projects explorer, you can browse your projects, view classpaths and dependencies. To get full feautures, you should create or open a folder containing Java projects.\n[Open Command Palette](command:workbench.action.showCommands)\n__Try searching for 'Create Java Project' or 'Open Folder'__",
```
Comment (nickzhums) [inline package.nls.json:6]: Java Project Explorer helps you to manage your projects and dependencies. Create a new project or open an existing folder to open the project explorer view

File: package.nls.json [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+    "walkthrough.java.debugAndTest":"The debugger will automatically find the entry point of your project, launch it and start debugging. [Learn More...](https://code.visualstudio.com/docs/java/java-debugging)\nThe [Testing View](command:testExplorer.focus) show all the test cases in your project. You can also run/debug your test cases from here. [Learn More...](https://code.visualstudio.com/docs/java/java-testing)\n[Show Run and Debug View](command:workbench.view.debug)",
+
+    "walkthrough.java.extensions": "Additional support for major toolchains, frameworks and application servers.\n[Browse Recommended Extensions](command:toSide:java.extGuide)",
```
Comment (nickzhums) [inline package.nls.json:10]: Install extensions to support additional frameworks such as Spring Boot, Application Servers, Keymaps and so on

File: package.nls.json [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{
+    "walkthrough.java.runtime": "Make sure you have at least on Java runtime installed.\n[Install New JDK](command:toSide:java.installJdk.fromWalkthrough)",
+
+    "walkthrough.java.projects": "In Java Projects explorer, you can browse your projects, view classpaths and dependencies. \n[View Java Projects](command:javaProjectExplorer.focus)",
+
+    "walkthrough.java.projects.inactive": "In Java Projects explorer, you can browse your projects, view classpaths and dependencies. To get full feautures, you should create or open a folder containing Java projects.\n[Open Command Palette](command:workbench.action.showCommands)\n__Try searching for 'Create Java Project' or 'Open Folder'__",
+
+    "walkthrough.java.debugAndTest":"The debugger will automatically find the entry point of your project, launch it and start debugging. [Learn More...](https://code.visualstudio.com/docs/java/java-debugging)\nThe [Testing View](command:testExplorer.focus) show all the test cases in your project. You can also run/debug your test cases from here. [Learn More...](https://code.visualstudio.com/docs/java/java-testing)\n[Show Run and Debug View](command:workbench.view.debug)",
```
Comment (nickzhums) [inline package.nls.json:8]: [Debug]
Run and debug your application using the debugger.
[Test]
Test Explorer allows you to see and manage all test cases in your project. You can also run tests directly from here

File: package.nls.json (hunk 8 above)
Comment (Eskibear) [reply package.nls.json:2]: >  Install new JDK from here

`[Install New JDK](command:toSide:java.installJdk.fromWalkthrough)` is a button occupying a whole line.

File: package.nls.json [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+
+    "walkthrough.java.extensions": "Additional support for major toolchains, frameworks and application servers.\n[Browse Recommended Extensions](command:toSide:java.extGuide)",
+
+    "walkthrough.java.more": "- Particularly if you want to run your project with a specific runtime, you can [Configure Java Runtime](command:toSide:java.runtime).\n\n- 🎉 Take a look at our [devblogs](akamsLinkToBlogs) for detailed release notes.\n\n- ✨ Discover more shortcuts and features in [Help Center](command:toSide:java.welcome.fromWalkthrough) or check Documentation for tips and troubleshooting."
```
Comment (nickzhums) [inline package.nls.json:12]: If you need to run your project with a specific Java runtime, you can [Configure Java Runtime]
Visit [Dev Blog] to keep up-to-date with our latest product updates and announcements
Discover more shortcuts and features at [Help Center]

Should we also add a link for Github repo to report issues?

File: walkthrough/runtime.md (hunk 6 above)
Comment (nickzhums) [reply walkthrough/runtime.md:3]: sounds good. let me think about it

File: package.nls.json (hunk 13 above)
Comment (Eskibear) [reply package.nls.json:12]: > Should we also add a link for Github repo to report issues?

Good idea.

File: package.nls.json (hunk 8 above)
Comment (nickzhums) [reply package.nls.json:2]: [Install JDK] - button text?

File: package.nls.json (hunk 13 above)
Comment (nickzhums) [reply package.nls.json:12]: - If you need to run your project with a specific Java runtime, you can [Configure Java Runtime]
- Visit [Dev Blog] to keep up-to-date with our latest product updates and announcements
- Discover more shortcuts and features at [Help Center]
- Report issues and feedback at [Github repo issues[

File: package.nls.json (hunk 8 above)
Comment (Eskibear) [reply package.nls.json:2]: OK I'll change `[Install New JDK]` to `[Install JDK]`.

**Comment by Eskibear** [comment]
TODO: 
- [ ] update markdown file for first step: install jdk 
- [ ] register aka.ms link for docs pointers

**Comment by Eskibear** [comment]
https://user-images.githubusercontent.com/2351748/145969156-63d5af6a-5605-4bd5-9680-0d11929dfdea.mp4

**Comment by nickzhums** [comment]
damn, looks good

**Comment by nickzhums** [comment]
btw - updating the markdown of first step to JDK 8 (instead of 11) is a good idea

**Comment by Eskibear** [comment]
@nickzhums updated.

**Comment by jdneo** [comment]
damn, looks good

**Comment by Eskibear** [comment]
damn, please approve

**Review by nickzhums — APPROVED** [review]

**Pull request merged**